modules/unpickler/src/main/scala/ch/epfl/scala/debugadapter/internal/stacktrace/Scala3Unpickler.scala

```diff
@@ -57,17 +59,39 @@ class Scala3Unpickler(
       case None =>
         throw new Exception(s"Cannot find Scala symbol of $fqcn")
       case Some(declaringType) =>
-        val matchingSymbols =
-          declaringType.declarations
-            .collect { case sym: TermSymbol if sym.isTerm => sym }
-            .filter(matchSymbol(method, _))
-
-        if matchingSymbols.size > 1 then
-          val message =
-            s"Found ${matchingSymbols.size} matching symbols for $method:" +
-              matchingSymbols.mkString("\n")
-          throw new Exception(message)
-        else matchingSymbols.headOption
+        matchesLocalMethodOrLazyVal(method) match
+          case Some((methodName, index)) => findLocalMethodOrLazyVal(declaringType, methodName, index)
+          case _ =>
+            val matchingSymbols = declaringType.declarations
+              .collect { case sym: TermSymbol if sym.isTerm => sym }
+              .filter(matchSymbol(method, _))
+            if matchingSymbols.size > 1 then
+              val message = s"Found ${matchingSymbols.size} matching symbols for $method:" +
+                matchingSymbols.mkString("\n")
+              throw new Exception(message)
+            else matchingSymbols.headOption
+
+  def findLocalMethodOrLazyVal(declaringType: DeclaringSymbol, name: String, index: Int): Option[TermSymbol] =
+    val matchingSymbols =
+      val declaringtpe = declaringType.owner match
+        case s: DeclaringSymbol => s
+        case _ => declaringType
+
+      declaringtpe.declarations
+        .flatMap(sym =>
+          sym.tree match
+            case Some(tree) =>
+              tree.walkTree(tree =>
+                tree match
+                  case DefDef(_, _, _, _, symbol) =>
+                    List((symbol, depth(declaringType, symbol)))
+                  case ValDef(_, _, _, symbol) => List((symbol, depth(declaringType, symbol)))
+                  case _ => List()
+              )((l1, l2) => l1 ++ l2, List())
+            case None => List()
+        )
+        .filter((symbol, depth) => matchTargetName(name, symbol) && depth >= 1)
+    Some(matchingSymbols.sortBy((_, depth) => depth).map((symbol, _) => symbol)(index - 1))
```

Comment:
The index is not the depth of the symbol but its order of appearance in the `declaringSymbol`. I don't think you need to compute the `depth` at all.

```suggestion
    Some(matchingSymbols(index - 1))
```

See, in my test, there is `m$1`,  `m$2` and `m$3`.